collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -58,9 +64,11 @@ public void doScrape() throws Exception {
         if (hostPort.isEmpty()) {
           beanConn = ManagementFactory.getPlatformMBeanServer();
         } else {
-          String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
-          jmxc = JMXConnectorFactory.connect(new JMXServiceURL(url), null);
-          beanConn = jmxc.getMBeanServerConnection();
+            String url = "service:jmx:rmi:///jndi/rmi://" + hostPort + "/jmxrmi";
+            HashMap credential = new HashMap();
+            credential.put(username, password );
```

Comment:
hm. I am afraid it is my eclipse configuration a little bit different. Could you send me your lang. style settings, then I'll alignment it to you, or in another hand, you can move out my mistakes by the merge.